### PR TITLE
Instrumentation events to LibrarieJS

### PR DESCRIPTION
**Pull request by ramramps**
### Purpose

This PR is to add Instrumentation to LibrarieJS.

I am not sure whether this is the right approach here. I am little scared to open the possibilities for all the view extensions to log to instrumentation (by adding events to extension manager). because analytics is a global singleton, it can be accessed if there is a reference to logging. so, I thought it's best way is to call instrumentation from html and handle the implementation in the view controller.
If someone feels this is incorrect, please let me know. we can improve this implementation. 

### Declarations

Check these if you believe they are true

- [ ] The code base is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions), and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.

### Reviewers
@alfarok @mjkkirschner @lukechurch

### CODE REVIEW

File: src/LibraryViewExtension/LibraryViewController.cs [hunk 1]
```diff
@@ -149,6 +152,18 @@ public void ImportLibrary()
                 dynamoViewModel.ImportLibraryCommand.Execute(null)
             ));
         }
+        /// <summary>
+        /// This function adds the events to instrumentation
```
Comment (mjkkirschner) [inline src/LibraryViewExtension/LibraryViewController.cs]: I don't really understand this comment, shouldn't this say something like logs the event if it matches a set of known events?

The function name doesn't make sense to me either - are there other examples of function names like this around logging and instrumentation? What are we adding instrumentation to?

File: src/LibraryViewExtension/LibraryViewController.cs [hunk 2]
```diff
@@ -149,6 +152,18 @@ public void ImportLibrary()
...
+        /// </summary>
+        /// <param name="eventName"> Event Name</param>
```
Comment (mjkkirschner) [inline src/LibraryViewExtension/LibraryViewController.cs]: these params don't tell us much.

File: src/LibraryViewExtension/LibraryViewController.cs [hunk 3]
```diff
@@ -137,6 +139,7 @@ public void CreateNode(string nodeName)
                 //Create the node of given item name
                 var cmd = new DynamoModel.CreateNodeCommand(Guid.NewGuid().ToString(), nodeName, -1, -1, true, false);
                 commandExecutive.ExecuteCommand(cmd, Guid.NewGuid().ToString(), ViewExtension.ExtensionName);
+                AddInstrumentation("Search-NodeAdded", nodeName);
```
Comment (mjkkirschner) [inline src/LibraryViewExtension/LibraryViewController.cs]: maybe we can pull out the magic strings and make them constants?

File: src/LibraryViewExtension/LibraryViewController.cs [hunk 4]
```diff
@@ -149,6 +152,18 @@ public void ImportLibrary()
...
             ));
         }
+        /// <summary>
+        /// This function adds the events to instrumentation
+        /// </summary>
+        /// <param name="eventName"> Event Name</param>
+        /// <param name="data">Data</param>
+        public void AddInstrumentation(string eventName, string data)
+        {
+            if (eventName == "Search" || eventName == "Filter-Categories" || eventName == "Search-NodeAdded")
```
Comment (mjkkirschner) [inline src/LibraryViewExtension/LibraryViewController.cs]: do you do this to try to prevent others from logging whatever they want via the library?

Comment (ramramps) [reply src/LibraryViewExtension/LibraryViewController.cs]: yes. I will add more comments.

File: src/LibraryViewExtension/LibraryViewController.cs (hunk 3 above)
Comment (ramramps) [reply src/LibraryViewExtension/LibraryViewController.cs]: sure

File: src/LibraryViewExtension/LibraryViewController.cs (hunk 1 above)
Comment (ramramps) [reply src/LibraryViewExtension/LibraryViewController.cs]: I am thinking more of `LogEventsToAnalytics` or `LogEventsToInstrumentation`. 
I couldn't find an example as analytics is global object.
thought @mjkkirschner ?

Comment (mjkkirschner) [reply src/LibraryViewExtension/LibraryViewController.cs]: `LogEventToInstrumentation` sounds good.

**Pull request merged**